pkg/virtctl/create/vm.go

```diff
@@ -0,0 +1,450 @@
+}
+
+func aliasNormalizeFunc(_ *pflag.FlagSet, flagName string) pflag.NormalizedName {
+	switch flagName {
+	case VolumeDvtPvcFlag:
+		flagName = VolumeClonePvcFlag
+	case VolumeDvtDsFlag:
+		flagName = VolumeCloneDsFlag
+	}
+	return pflag.NormalizedName(flagName)
+}
+
+func NewVirtualMachineCommand() *cobra.Command {
+	c := defaultCreateVM()
+	cmd := &cobra.Command{
+		Use:     VM,
+		Short:   "Create a VirtualMachine manifest.",
+		Example: c.usage(),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return c.run(cmd)
+		},
+	}
+
+	cmd.Flags().StringVar(&c.name, NameFlag, c.name, "Specify the name of the VM")
+	cmd.Flags().Int64Var(&c.terminationGracePeriod, "termination-grace-period", c.terminationGracePeriod, "Specify the termination grace period of the VM")
+
+	cmd.Flags().BoolVar(&c.running, RunningFlag, c.running, "Specify if the VM should be running")
+	cmd.Flags().StringVar(&c.runStrategy, RunStrategyFlag, c.runStrategy, "Specify the RunStrategy of the VM")
+	cmd.MarkFlagsMutuallyExclusive(RunningFlag, RunStrategyFlag)
+
+	cmd.Flags().StringVar(&c.instancetype, InstancetypeFlag, c.instancetype, "Specify the Instancetype of the VM")
+	cmd.Flags().StringVar(&c.preference, PreferenceFlag, c.preference, "Specify the Preference of the VM")
+
+	cmd.Flags().StringSliceVar(&c.volumeDvtPvc, VolumeClonePvcFlag, c.volumeDvtPvc, "Specify one or more PVCs to be cloned by the VM")
+	cmd.Flags().StringSliceVar(&c.volumeDvtDs, VolumeCloneDsFlag, c.volumeDvtDs, "Specify one or more DSs to be cloned by the VM")
+	cmd.Flags().StringSliceVar(&c.volumePvc, VolumePvcFlag, c.volumePvc, "Specify one or more PVCs to be used by the VM")
+	cmd.Flags().StringSliceVar(&c.volumeDv, VolumeDvFlag, c.volumeDv, "Specify one or more DVs to be used by the VM")
+
+	cmd.Flags().StringVar(&c.cloudInitUserData, CloudInitUserDataFlag, c.cloudInitUserData, "Specify the base64 encoded cloud-init user data of the VM")
+	cmd.Flags().StringVar(&c.cloudInitNetworkData, CloudInitNetworkDataFlag, c.cloudInitNetworkData, "Specify the base64 encoded cloud-init network data of the VM")
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	cmd.Flags().SetNormalizeFunc(aliasNormalizeFunc)
+
+	return cmd
+}
+
+func defaultCreateVM() createVM {
+	return createVM{
+		name:                   "vm-" + rand.String(5),
+		terminationGracePeriod: 180,
+	}
+}
+
+func (c *createVM) run(cmd *cobra.Command) error {
+	vm := newVM(c)
+	for _, flag := range flags {
+		if cmd.Flags().Changed(flag) {
+			if err := optFns[flag](c, vm); err != nil {
+				return err
+			}
+		}
+	}
+
+	out, err := yaml.Marshal(vm)
+	if err != nil {
+		return err
+	}
+
+	cmd.Print(string(out))
+	return nil
+}
+
+func (c *createVM) usage() string {
+	return `  # Create a VirtualMachine with a random name:
+  {{ProgramName}} create vm
+    
+  # Create a VirtualMachine with a specified name
+  {{ProgramName}} create vm --name=my-vm
+  
+  # Create a VirtualMachine with a specified Instancetype
```

Comment:
Can you update this to highlight these are `VirtualMachineCluster{Instancetype,Preference}s` and include a version that points to namespaced `VirtualMachine{Instancetype,Preference}s`?

Reply:
ACK